### ISSUE TITLE
Buttons aren't going active after CDs are up until after the GCD is complete

### ISSUE THREAD

**Issue by raysmith59** [brittyazel/Neuron#251]
Mid-raid so I can't spend a lot of time typing this up, I'll clarify it after if necessary.  Basically if a cooldown expires while the GCD is active then the button remains greyed out until the GCD is complete (or shortly after, it seems a bit laggy even there).  It's making it really hard to figure out if buttons are available or not.  There may be other things going on too but it's hard to figure out, I'll keep trying.

**Comment by brittyazel** [comment]
I think that might be the in game behavior though right? If an ability comes off CD during a GCD it should still be subject to the GCD. The only time this shouldn't be true is in the case for abilities not on the GCD, like Wild Charge, in which case Neuron shouldn't even be triggering the GCD on those buttons

**Comment by raysmith59** [comment]
Maybe in a literal sense but it's destroying my dps, my brain needs to queue up the next ability based on availability and right now I can't tell that.  I did figure out the other thing going on, not sure if I should report it separately.  Basically, if I queue up buttons (spamming, kinda) then the buttons NEVER activate after CDs expire so there is definitely some additional lag going on.  It's been a long time since I used the stock UI for bars but I don't believe they did either of these things.  The in-game GCD functionality will prevent buttons from being pushed if they aren't eligible yet due to GCD or otherwise, Neuron doesn't need to be involved in that.

I'll spend some time this weekend in a reverted build before the recent changes and confirm the previous behaviour.  Hang tight until then.

**Comment by brittyazel** [comment]
Cool. I'm happy to figure this out for you though. I don't think I play quitw on your level so I never noticed any lag

**Comment by raysmith59** [comment]
I didn't do a very good job of testing this scenario with the previous version but I'll document what I did find out for now, I'll provide an update after I re-test.

Something is awry between the timer expiration and the button being available.  I have an example below where the button has re-activated (lost the grey) but the timer is still showing 1, that never used to happen.  Here's the before and after.

So, here's an example of the timer being expired but the button not being clickable yet:
![image](https://user-images.githubusercontent.com/2917594/50743398-57d0de80-11e5-11e9-9828-881600612660.png)

I never had an issue with that behaviour and never felt that the button was clickable before it activated.

Now, the timer still shows after the button has re-activated:
![image](https://user-images.githubusercontent.com/2917594/50743414-9bc3e380-11e5-11e9-9610-688a06880cfe.png)

I was having trouble determining if the button could actually be clicked between re-activating and the timer expiring due to the command queuing that happens in-game so I disabled that functionality using AdvancedInterfaceOptions (by default it's set to 125ms):

![image](https://user-images.githubusercontent.com/2917594/50743497-c5c9d580-11e6-11e9-8ee3-74bc8bc7da90.png)

Once I disabled queuing it became clear that the button was NOT clickable when it activated, it was pretty easy for me to hit the button once it was available and have it not do anything.  I'm not 100% sure that the ending of the timer corresponds to when it was actually clickable but it's really close either way, closer than I'm capable of testing manually.

Long story short, it seems the button is re-activating too soon.  Note that this isn't what I reported here originally, I still need to test the impact of the GCD on this behaviour but this will need to be fixed regardless.

More soon.
